erizo/src/erizo/thread/Scheduler.cpp

```diff
@@ -29,8 +29,10 @@ void Scheduler::serviceQueue() {
         new_task_scheduled_.wait(lock);
       }
 
+      current_timeout_ = task_queue_.begin()->first;
       while (!stop_requested_ && !task_queue_.empty() &&
-             new_task_scheduled_.wait_until(lock, task_queue_.begin()->first) != std::cv_status::timeout) {
+             new_task_scheduled_.wait_until(lock, current_timeout_) != std::cv_status::timeout) {
+        boost::thread::yield();
```

Comment:
not sure why do we need to call `yield()` here?

Reply:
to give up our cpu time slice, since we are going to wait again